src/node_http2.h

```diff
@@ -501,6 +504,9 @@ class Http2Stream : public AsyncWrap,
   std::queue<NgHttp2StreamWrite> queue_;
   size_t available_outbound_length_ = 0;
 
+  // Outbound Headers
+  std::queue<std::unique_ptr<Http2Headers>> outgoing_headers_;
```

Comment:
Why unique_ptr, but not pointer?

Reply:
no good answer - the goal is to ensure that underlaying `Http2Headers` class memory is not freed until the frame has been sent out, due to an addition of NO_COPY flags for underlaying char pointers - they are being referenced by nghttp2 until the callback is called. 

if there is a better way to manage memory from being deallocated - would happily change that